### ISSUE TITLE
Add defined-types-are-used rule

### ISSUE THREAD

**Issue by cjoudrey** [cjoudrey/graphql-schema-linter#20]
In order to ensure old types get removed when they are no longer used, we could have a rule that verifies that all types in the schema are referenced at least once.

For example, the rule would warn about the type `User` not being used in the following schema:

```graphql
type QueryRoot {
  a: String!
}

type User {
  email: String!
}
```

**Comment by cjoudrey** [comment]
@goldcaddy77 any thoughts on this one?

**Comment by goldcaddy77** [comment]
Love it... would be very similar to this, right? https://github.com/graphql/graphql-js/blob/master/src/validation/rules/NoUnusedFragments.js

**Comment by cjoudrey** [comment]
Yeah I think the code would be similar.

Something like:

```
definedTypes = []
referencedTypes = []

ObjectTypeDefinition: (node) {
  definedTypes << node.name.value
}

NamedType: (node) {
  referencedTypes << node.name.value
}

Document: {
  leave(node) {
    // Verify that definedTypes == referencedTypes
  }
}
```
